File: Competitions/contest-121/984-不含AAA或BBB的字符串/984.java
class Solution {
    public String strWithout3a3b(int A, int B) {
        String sb = new String();
        do {
            if (A >= B) { //a多，尽量先消费a（贪心策略）
                if (sb.endsWith("aa")) { //实在不能消耗a才消费b
                    sb = sb.concat("b");
                    B--;
                } else { //先消费a
                    sb = sb.concat("a");
                    A--;
                }
            } else {
                if (sb.endsWith("bb")) {
                    sb = sb.concat("a");
                    A--;
                } else {
                    sb = sb.concat("b");
                    B--;
                }
            }
        } while (A>0 || B>0); //只要A或B不等于0，则一直循环
        return sb;
    }
}

File: Competitions/contest-121/984-不含AAA或BBB的字符串/984.cpp
class Solution {
public:
    bool endsWith(string s, string sub){
        return s.rfind(sub) == (s.length()-sub.length())&&(s.length()>=sub.length())?true:false;
    }
    string strWithout3a3b(int A, int B) {
        string res;
        while(A>0 || B>0){
            if (A >= B) {
                if(endsWith(res,"aa")){
                    res+="b";
                    B-=1;
                }
                else{
                    res+="a";
                    A-=1;
                }
            } else {
                if(endsWith(res,"bb")){
                    res+="a";
                    A-=1;
                }
                else{
                    res+="b";
                    B-=1;
                }
            }
        }
        return res;
    }
};